crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp

```diff
@@ -10,28 +10,59 @@ crp::apl::ControlHandler::ControlHandler() : Node("ControlHandler")
 
-    m_sub_controlLat_ = this->create_subscription<autoware_control_msgs::msg::Control>("/control/command/control_cmdLat", 10, std::bind(&ControlHandler::controlLatCallback, this, std::placeholders::_1));
-    m_sub_controlLong_ = this->create_subscription<autoware_control_msgs::msg::Control>("/control/command/control_cmdLong", 10, std::bind(&ControlHandler::controlLongCallback, this, std::placeholders::_1));
+    m_sub_controlLat_ = this->create_subscription<autoware_control_msgs::msg::Lateral>("/control/command/control_cmdLat", 10, std::bind(&ControlHandler::controlLatCallback, this, std::placeholders::_1));
+    m_sub_controlLong_ = this->create_subscription<autoware_control_msgs::msg::Longitudinal>("/control/command/control_cmdLong", 10, std::bind(&ControlHandler::controlLongCallback, this, std::placeholders::_1));
 
     RCLCPP_INFO(this->get_logger(), "ctrl_vehicle_control has been started");
+
+    // initialize control message
+    ctrl_msg.lateral.steering_tire_angle = 0.0f;
+    ctrl_msg.lateral.steering_tire_rotation_rate = 0.0f;
+    ctrl_msg.longitudinal.velocity = 0.0f;
+
+    // initialize twist message
+    m_twistMsg.linear.x = 0.0f;
+    m_twistMsg.linear.y = 0.0f;
+    m_twistMsg.linear.z = 0.0f;
+    m_twistMsg.angular.x = 0.0f;
+    m_twistMsg.angular.y = 0.0f;
+    m_twistMsg.angular.z = 0.0f;
+
 }
 
-void crp::apl::ControlHandler::controlLatCallback(const autoware_control_msgs::msg::Control::SharedPtr msg)
+void crp::apl::ControlHandler::controlLatCallback(const autoware_control_msgs::msg::Lateral::SharedPtr msg)
 {
-    twist_msg.angular.z = msg->lateral.steering_tire_angle;
-    ctrl_msg.lateral.steering_tire_angle = msg->lateral.steering_tire_angle;
-    ctrl_msg.lateral.steering_tire_rotation_rate = 0.0f;
+    // check for nan values
+    if (std::isnan(msg->steering_tire_angle))
+    {
+        RCLCPP_WARN(this->get_logger(), "Received nan value for steering angle");
+        return;
+    }
+
+    m_twistMsg.angular.z = msg->steering_tire_angle;
+    ctrl_msg.lateral.steering_tire_angle = msg->steering_tire_angle;
+    ctrl_msg.lateral.steering_tire_rotation_rate = msg->steering_tire_rotation_rate;
 }
 
-void crp::apl::ControlHandler::controlLongCallback(const autoware_control_msgs::msg::Control::SharedPtr msg)
+void crp::apl::ControlHandler::controlLongCallback(const autoware_control_msgs::msg::Longitudinal::SharedPtr msg)
 {
-    twist_msg.linear.x = msg->longitudinal.velocity;
-    ctrl_msg.longitudinal.velocity = msg->longitudinal.velocity;
+    // check for nan values
+    if (std::isnan(msg->velocity))
+    {
+        RCLCPP_WARN(this->get_logger(), "Received nan value for velocity");
+        return;
+    }
+
+
+    m_twistMsg.linear.x = msg->velocity;
+    ctrl_msg.longitudinal.velocity = msg->velocity;
 }
 
 void crp::apl::ControlHandler::run()
 {
-    m_pub_twist_->publish(twist_msg);
+    ctrl_msg.stamp = this->now();
```

Comment:
shall we add the same stamp to the twist message as well?

Reply:
the twist msg has no time stamp